numalogic/registry/dynamodb_registry.py

```diff
@@ -0,0 +1,398 @@
+from numalogic.tools.exceptions import DynamoDBRegistryError
+from numalogic.tools.types import KEYS, artifact_t, META_VT
+from numalogic.registry._serialize import loads, dumps
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AWSConnector:
+    """
+    Class to connect to any AWS resource.
+
+    Args:
+        role: AWS role
+        region: AWS region (default: us-west-2)
+    """
+
+    sess_start_time = None
+    session = None
+    __STS_RESET_SECS = 30 * 60
+
+    def __init__(self, role=None, region="us-west-2"):
+        self.role = role
+        self.region = region
+
+    def get_session(self, new_session=False) -> boto3.Session:
+        if new_session:
+            self.session = None
+
+        if (not self.session) or (time.time() - self.sess_start_time > self.__STS_RESET_SECS):
+            self.session = self._get_sts_session()
+            self.sess_start_time = time.time()
+        return self.session
+
+    def _get_sts_session(self):
+        _LOGGER.info("Getting session with role: %s and region: %s", self.role, self.region)
+        assumed_role_object = boto3.client("sts").assume_role(
+            RoleArn=self.role, RoleSessionName="AssumeRoleSession1"
+        )
+        return boto3.session.Session(
+            aws_access_key_id=assumed_role_object["Credentials"]["AccessKeyId"],
+            aws_secret_access_key=assumed_role_object["Credentials"]["SecretAccessKey"],
+            aws_session_token=assumed_role_object["Credentials"]["SessionToken"],
+            region_name=self.region,
+        )
+
+    def get_client(self, service="dynamodb", **kwargs):
+        return self.get_session(**kwargs).client(service)
+
+    def get_resource(self, service="dynamodb", **kwargs):
+        return self.get_session(**kwargs).resource(service)
+
+
+class DynamoDBRegistry(ArtifactManager):
+    """
+    Model saving and loading to and from Dynamodb.
+    Args:
+        table: table name to use
+        connector: AWSConnector instance
+        models_to_retain: number of models to retain in the DB (default = 2)
+        cache_registry: ArtifactCache instance, must have an expiration set for the
+         model to be refreshed
+    Examples
+    --------
+    >>> import redis
+    >>> from numalogic.models.autoencoder.variants import VanillaAE
+    >>> from numalogic.registry.dynamodb_registry import DynamoDBRegistry, AWSConnector
+    >>> ...
+    >>> connector = AWSConnector(role="arn:aws:iam::1234567890:role/role-name")
+    >>> registry = DynamoDBRegistry(table="mytable", connector=connector)
+    >>> skeys, dkeys = ("mymetric", "ae"), ("vanilla", "seq10")
+    >>> model = VanillaAE(seq_len=10)
+    >>> registry.save(skeys, dkeys, artifact=model, **{'lr': 0.01})
+    >>> loaded_artifact = registry.load(skeys, dkeys).
+    """
+
+    __slots__ = ("table_name", "connector", "models_to_retain", "cache_registry")
+
+    def __init__(
+        self,
+        table: str,
+        connector: AWSConnector,
```

Comment:
```suggestion
        role: str,
        region: Optional[str] = None
```
Let's maybe create the awsconnector instance internally here. This helps in 2 ways:
- No need to initialize and import awsconnector separately
- Easier initialization via config